monkey/monkey_island/cc/setup/island_config_options.py

```diff
@@ -20,22 +19,39 @@
 
 
 class IslandConfigOptions:
-    def __init__(self, config_contents: dict = None):
-        if not config_contents:
-            config_contents = {}
-        self.data_dir = config_contents.get(_DATA_DIR, DEFAULT_DATA_DIR)
+    def __init__(self, config_contents: dict = {}):
+        self.data_dir = None
+        self.log_level = None
+        self.start_mongodb = None
+        self.crt_path = None
+        self.key_path = None
+
+        self._initialize_defaults()
+        self.update(config_contents)
+
+    def _initialize_defaults(self):
+        self.data_dir = DEFAULT_DATA_DIR
+        self.log_level = DEFAULT_LOG_LEVEL
+        self.start_mongodb = DEFAULT_START_MONGO_DB
+        self.crt_path = DEFAULT_CRT_PATH
+        self.key_path = DEFAULT_KEY_PATH
 
-        self.log_level = config_contents.get(_LOG_LEVEL, DEFAULT_LOG_LEVEL)
+        self._expand_paths()
+
+    def update(self, config_contents: dict):
+        self.data_dir = config_contents.get(_DATA_DIR, self.data_dir)
+
+        self.log_level = config_contents.get(_LOG_LEVEL, self.log_level)
 
         self.start_mongodb = config_contents.get(
-            _MONGODB, {_START_MONGODB: DEFAULT_START_MONGO_DB}
-        ).get(_START_MONGODB, DEFAULT_START_MONGO_DB)
+            _MONGODB, {_START_MONGODB: self.start_mongodb}
+        ).get(_START_MONGODB, self.start_mongodb)
 
-        self.crt_path = config_contents.get(_SSL_CERT, DEFAULT_CERTIFICATE_PATHS).get(
-            _SSL_CERT_FILE, DEFAULT_CRT_PATH
+        self.crt_path = config_contents.get(_SSL_CERT, {_SSL_CERT_FILE: self.crt_path}).get(
+            _SSL_CERT_FILE, self.crt_path
         )
-        self.key_path = config_contents.get(_SSL_CERT, DEFAULT_CERTIFICATE_PATHS).get(
-            _SSL_CERT_KEY, DEFAULT_KEY_PATH
+        self.key_path = config_contents.get(_SSL_CERT, {_SSL_CERT_KEY: self.key_path}).get(
+            _SSL_CERT_KEY, self.key_path
```

Comment:
This was the problem? paths not expanded and expected object instead of string?

Reply:
Yeah. This object is flat, whereas the config file is not. So using `self.__dict__.update()` won't work.